tests/IntegrationTest.php

```
@@ -0,0 +1,479 @@
+    }
+
+    public function testConcurrentAccessSimulation(): void
+    {
+        $sharedData = range(1, 100);
+        $accessLog = [];
+
+        $source = new SourceCallbackAdapter(function (int $page, int $size) use ($sharedData, &$accessLog) {
+            $accessLog[] = ['page' => $page, 'size' => $size, 'time' => microtime(true)];
+
+            $startIndex = ($page - 1) * $size;
+            $endIndex = min($startIndex + $size, count($sharedData));
```

PHPMD (2.15.0)

Avoid unused local variables such as '$endIndex'. (undefined)

(UnusedLocalVariable)